install_config/advanced_ldap_configuration/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,425 @@
+  SSLProxyEngine on
+  SSLProxyCACertificateFile /etc/pki/CA/certs/ca.crt
+  # It's critical to enforce client certificates on the Master.  Otherwise
+  # requests could spoof the X-Remote-User header by accessing the Master's
+  # /oauth/authorize endpoint directly.
+  SSLProxyMachineCertificateFile /etc/pki/tls/certs/authproxy.pem
+
+  # Send all requests to the console
+  RewriteEngine              On
+  RewriteRule     ^/console(.*)$     https://%{HTTP_HOST}:8443/console$1 [R,L]
+
+  # In order to using the challenging-proxy an X-Csrf-Token must be present.
+  RewriteCond %{REQUEST_URI} ^/challenging-proxy
+  RewriteCond %{HTTP:X-Csrf-Token} ^$ [NC]
+  RewriteRule ^.* - [F,L]
+
+  <Location /challenging-proxy/oauth/authorize>
+    # Insert your backend server name/ip here.
+    ProxyPass https://openshift.example.com:8443/oauth/authorize
+    AuthType Basic
+    AuthBasicProvider PAM
+    AuthPAMService openshift
+    Require valid-user
+  </Location>
+
+  <ProxyMatch /oauth/authorize>
+    AuthName openshift
+    RequestHeader set X-Remote-User %{REMOTE_USER}s
+  </ProxyMatch>
+</VirtualHost>
+
+RequestHeader unset X-Remote-User
+----
+====
++
+[NOTE]
+====
+link:../advanced_ldap_configuration/configuring_form_based_authentication.html[Configuring
+Form-Based Authentication] explains how to add the *login-proxy* block to
+support form authentication.
+====
+
+. Set a boolean to tell SELinux that it is acceptable for Apache to contact the
+PAM subsystem:
++
+====
+----
+# setsebool -P allow_httpd_mod_auth_pam on
+----
+====
+
+. Start up Apache:
++
+====
+----
+# systemctl start httpd.service
+----
+====
+
+[[sssd-phase-3-openshift-configuration]]
+== Phase 3: {product-title} Configuration
+
+This section describes how to set up an {product-title} server from scratch in
+an "all in one" configuration.
+link:../master_node_configuration.html[Master and Node
+Configuration] provides more information on alternate configurations.
+
+Modify the default configuration to use the new identity provider just
+created. To do so:
+
+. Modify the *_/etc/origin/master/master-config.yaml_* file.
+
+. Scan through it and locate the *identityProviders* section and replace it with:
++
+====
+----
+  identityProviders:
+  - name: any_provider_name
+    challenge: true
+    login: false
+    mappingMethod: claim
+    provider:
+      apiVersion: v1
+      kind: RequestHeaderIdentityProvider
+      challengeURL: "https://proxy.example.com/challenging-proxy/oauth/authorize?${query}"
+      clientCA: /etc/origin/master/proxy/proxyca.crt
+      headers:
+      - X-Remote-User
+----
+====
++
+[NOTE]
+====
+link:../advanced_ldap_configuration/configuring_form_based_authentication.html[Configuring
+Form-Based Authentication] explains how to add the login URL to support web
+logins.
+
+link:../advanced_ldap_configuration/configuring_extended_ldap_attributes.html[Configuring
+Extended LDAP Attributes] explains how to add the email and full-name
+attributes.
+====
```

Comment:
The more I think about this, the more I suspect these should all just be merged together. One thing I learned when developing my blog entries was that the full-name is only stored to the database on the first login (which is probably a bug, but that's the way of it right now). So if we have people run through these in this sequence, the users they initially test with are probably going to be missing that full-name. (This can be fixed by deleting the user, but that's a destructive solution).

So I guess I'm switching my vote to merging these all together and doing it in one go.